com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ModelUtil.java

```diff
@@ -670,48 +670,56 @@ private static void findModelNodes(Node node, List<Node> modelNodes) {
         }
     }
 
+    private static void calcCenterNode(Node node, Aabb aabb) {
+        if (node.model != null) {
+            // As a default, we only count nodes with models, as the user
+            // cannot currently see/use the lights or cameras etc that are present in the scene.
+            aabb.expand(node.world.translation.x, node.world.translation.y, node.world.translation.z);
+        }
+
+        for (Node child : node.children) {
+            calcCenterNode(child, aabb);
+        }
+    }
+
+    // Currently finds the center point using the world positions of each node
     private static ModelImporter.Vec4 calcCenter(Scene scene) {
-        ModelImporter.Vec4 center = new ModelImporter.Vec4(0.0f, 0.0f, 0.0f, 0.0f);
-        float count = 0.0f;
+        Aabb aabb = new Aabb();
         for (Node root : scene.rootNodes) {
-            ArrayList<Node> modelNodes = new ArrayList<>();
-            findModelNodes(root, modelNodes);
-
-            for (Node modelNode : modelNodes) {
-                center.x += modelNode.local.translation.x;
-                center.y += modelNode.local.translation.y;
-                center.z += modelNode.local.translation.z;
-                count++;
-            }
+            calcCenterNode(root, aabb);
         }
-        center.x /= (float)count;
-        center.z /= (float)count;
-        center.x /= (float)count;
+
+        ModelImporter.Vec4 center = new ModelImporter.Vec4(0.0f, 0.0f, 0.0f, 0.0f);
+        if (aabb.isValid())
+            center = aabb.center();
         return center;
     }
 
+    private static void shiftModelsNode(Node node, ModelImporter.Vec4 center) {
+        node.world.translation.x -= center.x;
+        node.world.translation.y -= center.y;
+        node.world.translation.z -= center.z;
+
+        for (Node child : node.children) {
+            shiftModelsNode(child, center);
+        }
+    }
```

Comment:
Once the center of the aabb is calcualted, we can shift the scene there.
In this case we move _all_ the nodes, as we want their relative positions intact.